sbws/core/scanner.py

```diff
@@ -354,6 +354,37 @@ def closure(object):
     return closure
 
 
+def main_loop(args, conf, controller, relay_list, circuit_builder, result_dump,
+              relay_prioritizer, destinations, max_pending_results, pool):
+    pending_results = []
+    while not settings.end_event.is_set():
+        num_relays = 0
+        loop_tstart = time.time()
+        for target in relay_prioritizer.best_priority():
+            if settings.end_event.is_set():
+                break
+            num_relays += 1
+            log.debug('Measuring %s %s', target.nickname,
+                      target.fingerprint[0:8])
+            callback = result_putter(result_dump)
+            callback_err = result_putter_error(target)
+            async_result = pool.apply_async(
+                dispatch_worker_thread,
+                [args, conf, destinations, circuit_builder, relay_list,
+                 target], {}, callback, callback_err)
+            pending_results.append(async_result)
+            log.debug(async_result)
+            while len(pending_results) >= max_pending_results:
+                time.sleep(5)
```

Comment:
This is not part of this new patch, but what's the point of sleeping for 5 seconds?

Reply:
Every `async_result` has its own thread and the thread are reused. I assumed that was needed to don't reuse a thread before the `async_result` is finished, but maybe it's not need.

Reply:
Hmm, I'm not sure if sleeping is the right way to prevent anything in multi-threading code but I also don't understand exactly what is meant to be achieved. Perhaps adding a comment about it, might help us remember it later if we want to discuss it in Brussels?

Also, should we add function documentation to this new main_loop function?

Reply:
Added in e4e671d. Maybe your questions are now answered with the documentation?.